lib/rpmrc.cc

```diff
@@ -358,6 +361,109 @@ const char * lookupInDefaultTable(const char * name,
     return name;
 }
 
+static int moveConfigFiles(std::string userdir)
+{
+    /* Don't migrate in scripts */
+    if (!isatty(STDIN_FILENO))
+	return -1;
+    /* Don't migrate with root priviledges */
+    if (geteuid() == 0)
+	return -1;
+
+    fs::path home = std::getenv("HOME");
+    std::error_code ec;
+    std::error_code ec2;
+    fs::path oldmacros = home / ".rpmmacros";
+    fs::path oldrpmrc = home / ".rpmrc";
+    fs::path lockfile = home / ".rpmconfig_migration_lock";
+    fs::path macros_target = "";
+    fs::path rpmrc_target = "";
+
+    bool move_rpmrc = false;
+    bool move_macros = false;
+
+    if (userdir.substr(0, 2) == "~/")
+	userdir.replace(0, 1, home);
+
+    fs::path userdir_path = userdir;
+    fs::path newmacros = userdir_path / "macros";
+    fs::path newrpmrc = userdir_path / "rpmrc";
+
+    if (!fs::is_regular_file(oldmacros, ec) && !fs::is_regular_file(oldrpmrc, ec))
+	return -1;
+
+    int fd = open(lockfile.c_str(), O_CREAT|O_EXCL|O_WRONLY, 0644);
```

Comment:
Perhaps silly, but couldn't we reuse the `rpmlock` mechanism here? That said, I'm not familiar with the inner workings of it myself, so not sure it's a good fit, just something that comes to mind.